### DISCUSSION TITLE
how to change return table (array of records) in rpc?

### DISCUSSION THREAD

**Discussion by tatiana1337** [supabase/supabase#5008] [Questions]
which return type should I choose to get array of records (table) from rpc? 

```
    let { data, error } = await supabase
        .rpc('get_posts')
```
this block of code returns only 1 single record from table posts

`
SELECT * from posts`

**Comment by dshukertjr** [comment]
Hi @tatiana1337 !

Did you create the SQL function with raw SQL or from the Supabase console? Do you happen to have the definition of your `get_posts` function?

**Reply by tatiana1337** [reply]
> Hi @tatiana1337 !
> 
> Did you create the SQL function with raw SQL or from the Supabase console? Do you happen to have the definition of your `get_posts` function?

hello! @dshukertjr,  thanks for the reply
I have created this function in supabase console (not SQL editor, but directly in Database/Functions block) like this:
`SELECT * from posts` with return type "record" because there is nothing like "table" or "array" and json also didn't work

what is the proper way to add this rpc? 🙂

**Comment by steve-chavez** [comment]
The function should be declared with `SETOF`, like:

```sql
create or replace function get_posts() returns setof posts as $$
  select * from posts
$$ language sql
```
